crates/lib/src/executable.rs

```diff
@@ -587,6 +587,20 @@ impl<'execution> Executable<'_, 'execution> {
         Ok(job_handle)
     }
 
+    /// Cancel a job that has yet to begin executing.
+    ///
+    /// This action is *not* atomic, and will attempt to cancel a job even if it cannot be cancelled. A
+    /// job can be cancelled only if it has not yet started executing.
+    ///
+    /// Success response indicates only that the request was received. Cancellation is not guaranteed,
+    /// as it is based on job state at the time of cancellation, and is completed on a best effort
+    /// basis.
+    pub async fn cancel_qpu_job(&mut self, job_handle: JobHandle<'execution>) -> Result<(), Error> {
```

Comment:
Since cancellation is not guaranteed, I feel like the name could clarify this.

```suggestion
    pub async fn request_qpu_job_cancellation(&mut self, job_handle: JobHandle<'execution>) -> Result<(), Error> {
```